examples/ensemble/plot_ensemble_oob.py

```diff
@@ -0,0 +1,91 @@
+"""
+=========================================================
+OOB Errors for Random Forests and Extra Trees Classifiers
+=========================================================
+
+The ``RandomForestClassifier`` and ``ExtraTreesClasifier`` are trained using
+*bootstrap aggregation*. During training, each new tree is fit from a
+bootstrap sample of the training observations :math:`z_i = (x_i, y_i)`. The
+*out-of-bag* (OOB) error is the average prediction error for each :math:`z_i`
+from trees that do not contain :math:`z_i` in their respective bootstrap
+sample. This allows models to be simultaneously fit and cross-validated [1].
```

Comment:
p593 of Hastie et al. states:

> Hence unlike many other nonlinear estimators,
random forests can be fit in one sequence, with cross-validation being per-
formed along the way.

while not a standard form of cross-validation, are the OOB errors technically not estimated via type of cross-validation? However, I'm happy to change to "validation" to avoid potential confusion with something like k-fold cross validation.